File: jas-samples/sast/java/Vuln.java
// JAS SAST demo fixture (Java). Not part of any build (no build file references it).
//
// Findings demonstrated:
//  - CWE-611 / OWASP A05:2021 Security Misconfiguration: XXE via a
//    DocumentBuilderFactory that has not disabled external entities (Medium).
//  - CWE-502 Deserialization of untrusted data: ObjectInputStream reading
//    attacker-controlled bytes (High).
//  - CWE-798 Use of hard-coded credentials (Low/Medium).

import java.io.*;
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import org.w3c.dom.Document;

public class Vuln {

    private static final String DB_PASSWORD = "adminP@ssw0rd!"; // Vulnerable: hardcoded credential

    // Vulnerable: DocumentBuilderFactory built with default settings allows
    // external entity resolution (XXE), enabling file disclosure / SSRF.
    public Document parseUntrustedXml(InputStream untrustedXml) throws Exception {
        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
        DocumentBuilder builder = dbf.newDocumentBuilder();
        return builder.parse(untrustedXml);
    }

    // Vulnerable: deserializing an attacker-controlled stream can lead to
    // remote code execution via gadget chains.
    public Object loadSession(InputStream untrustedInput) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(untrustedInput);
        return ois.readObject();
    }
}

File: jas-samples/sast/rust/vuln.rs
// JAS SAST demo fixture (Rust). Not part of any crate/build in this repo.
//
// Findings demonstrated:
//  - CWE-502 Deserialization of untrusted data: deserializing an
//    attacker-controlled byte stream with no type/size validation (Medium).
//  - CWE-798 Use of hard-coded credentials (Low/Medium).

use serde::Deserialize;

const API_TOKEN: &str = "rust-demo-hardcoded-token-not-real"; // Vulnerable: hardcoded credential

#[derive(Deserialize)]
struct SessionPayload {
    user_id: String,
    is_admin: bool,
}

// Vulnerable: deserializes attacker-supplied bytes straight into a struct
// that includes a privilege field (`is_admin`), with no source
// authentication or schema/allowlist validation upstream.
fn load_session(untrusted_bytes: &[u8]) -> Result<SessionPayload, serde_json::Error> {
    serde_json::from_slice(untrusted_bytes)
}

fn authenticate(token: &str) -> bool {
    token == API_TOKEN
}
